1.fee_bumping/fee_bumping.md

```diff
@@ -153,25 +153,217 @@ Bitcoin blockchain, such as exchanges or custodians:
+  the inputs from the original transaction. This can be extremely dangerous
+  since wallets can easily overspend this way. For example, if the original
+  transaction uses inputs A and B, a replacement transaction is sent with
+  inputs B and C, and then that replacement transaction is replaced with a
+  new replacement transaction using inputs C and D, the first transaction
+  could be rebroadcast by a third party at any point in the future, and both
+  would be valid! Wallets are advised to include all inputs from the original
+  transaction in the replacement transaction unless they have very reliable
+  tracking to make sure that they don't double-spend themselves.
+- implication 4: the `nSequence` of the inputs in the replacement transaction
+  are not considered, so a replaceable transaction could be replaced by a
+  non-replaceable transaction. If a wallet wants to be able to replace the
+  replacement transaction in future, it must signal opt-in RBF on one of the
+  inputs of the replacement transaction.
+
+> 1. The original transactions signal replaceability explicitly or through
+> inheritance as described in the above Summary section.
+
+- implication 1: It is not sufficient to look at an individual transaction
+  to determine whether it can be replaced under this policy. A transaction
+  may be replaceable because one of its unconfirmed ancestors signals
+  opt-in RBF.
+- implication 2: A transaction's replaceability may change! If it is does
+  not signal opt-in replaceability, but one of its ancestors does, then it
+  is replaceable. If the ancestor then gets confirmed, the descendant
+  transaction becomes non-replaceable.
+
+> 2. The replacement transaction does not contain any new unconfirmed inputs
+> that did not previously appear in the mempool. (Unconfirmed inputs are
+> inputs spending outputs from currently unconfirmed transactions.)
+
+Rationale: this is an anti-DoS measure to stop a spender creating a replacement
+transaction which has a higher feerate than the original transaction, but
+adds new ancestor dependencies so that the package feerate for the transaction
+is lower.
+
+- implication 1: if a wallet runs out of confirmed transactions (ie all of its
+  coins are tied up in unconfirmed transactions), then it can't use the output
+  of some of those transactions as additional inputs to bump the fee on its other
+  transactions.
 
-### 5 conditions for RBF
-
-TODO
+> 3. The replacement transaction pays an absolute fee of at least the sum paid
+> by the original transactions.
+
+- implication 1: if outputs from the original transaction have been spent, then
+  the replacement transaction must exceed both the *feerate* and *total fee* of
+  the original transaction and its descendants. If the child transaction is large,
+  or there are many descendant transactions, then the total fee required to
+  replace the original transaction could be very large. This is known as 
+  *transaction pinning*.
```

Comment:
I think this last sentence could use just a few more words for context, e.g.: "This can make it uneconomical to replace the original transaction and so is known as *transaction pinning.*"

Reply:
Done!